src/transformStateWithClones.js

```diff
@@ -7,7 +7,52 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+
+  function removeProperties(object, keysToRemove) {
+    for (const key of keysToRemove) {
+      delete object[key];
+    }
+  }
+
+  function clearObject(object) {
+    for (const key in object) {
+      delete object[key];
+    }
+  }
+
+  function addProperties(object, propertiesToAdd) {
+    Object.assign(object, propertiesToAdd);
+  }
```

Comment:
Maybe declare these functions outside of `transformStateWithClones` because it is too big. What do you think?